packages/runtime/src/widgets/Form/TextInput.tsx

```diff
@@ -21,6 +21,11 @@ export const TextInput: React.FC<TextInputProps> = (props) => {
   const { values } = useRegisterBindings({ ...props, value }, props.id, {
     setValue,
   });
+  const form = Form.useFormInstance();
+
+  useEffect(() => {
+    form.setFieldsValue({ [values?.id ?? values?.label ?? ""]: value });
+  }, [value]);
```

Comment:
We might not always be in a Form instance...this should still work? Why not just call setValue and then onChange?